packages/apps/autorest/src/args.ts

```diff
@@ -0,0 +1,83 @@
+import { homedir } from "os";
+import { join } from "path";
+
+export interface AutorestArgs {
+  // Versioning
+  "v3"?: boolean;
+  "preview"?: boolean;
+  "prerelease"?: boolean;
+  "version"?: string;
+  "latest"?: boolean;
+
+  "reset"?: boolean;
+  "debug"?: boolean;
+  "info"?: boolean;
+  "json"?: boolean;
+  "configFileOrFolder"?: string;
+  "force"?: boolean;
+
+  "message-format"?: "regular" | "json" | "yaml";
+  "list-available"?: boolean;
+  "clear-temp"?: boolean;
+  "list-installed"?: boolean;
+  "skip-upgrade-check"?: boolean;
+}
+
+/**
+ * Regex to parse flags in format:
+ * --foo:bar
+ * --foo=bar
+ * --foo
+ */
+const FLAG_REGEX = /^--([^=:]+)([=:](.+))?$/;
+
+/**
+ * Parse a list of command line arguments.
+ * @param argv List of cli args(process.argv)
+ */
+export const parseArgs = (argv: string[]): AutorestArgs => {
```

Comment:
Seems like autorest core is doing more or less the same thing. Might be good to share the code.